cloud/scope/powervs_cluster.go

```diff
@@ -1899,6 +1903,7 @@ func (s *PowerVSClusterScope) ReconcileLoadBalancers() (bool, error) {
 			return false, err
 		}
 		if loadBalancerStatus != nil {
+			s.V(3).Info("Found VPC load balancer in IBM Cloud", "id", *loadBalancerStatus.ID)
 			s.SetLoadBalancerStatus(loadBalancer.Name, *loadBalancerStatus)
```

Comment:
I think here loadBalancer.Name will not be set properly if its nil. Scope of loadBalancer in line 1880 will end after if block.
You also added check loadBalancer.ID == nil && loadBalancer.Name there will be an issue when id is not nil but name is nil.

Reply:
The loadBalancer variable here is referring to this right? https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/1f473c2bc73e3daadd4a2c0747d462ef8e8bbccd/cloud/scope/powervs_cluster.go#L1867 

and the flow will reach here when resource is checked via given name and it will be set in either case to the user provided name or default name as per the flow

> You also added check loadBalancer.ID == nil && loadBalancer.Name there will be an issue when id is not nil but name is nil.

can you elaborate more on this?

Reply:
Can you check the flow when loadbalancer.id is set and loadbalancer.name is nil?

Reply:
When loadbalancer.ID is set we check the resource with given ID in cloud and set the status [here](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/pull/1937/files#diff-a12914f75cce0ea4c617f28f71fcfbd158441b44cb39595fe88f16c69c269b79R1897)